earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 The Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <time.h>
+
+#include "common/performancelogger.h"
+#include "common/notify.h"
+#include "common/timeutils.h"
+
+using namespace std;
+using namespace getime;
+
+// Initialize static members of Profiler class
+PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+
+// Initialize static members of BlockProfiler class
+PerformanceLogger * const BlockPerformanceLogger::perfLogger = PerformanceLogger::instance();
+
+// Log a profiling message
+void PerformanceLogger::log(
+    const string & operation,
+    const string & object,
+    const timespec startTime,
+    const timespec endTime,
+    const size_t size) {
+
+  const timespec duration = timespecDiff(endTime, startTime);
+  const pid_t tid = syscall(SYS_gettid);
+  stringstream message;
+
+  message.setf(ios_base::fixed, ios_base::floatfield);
+  message << setprecision(9)
+          << operation << " " << object << ": "
+          << "start time: " << startTime
+          << ", "
+          << "end time: " << endTime
+          << ", "
+          << "duration: " << duration
```

Comment:
may be add unit for duration message. seconds i gather ? later we may want this to have more smart into it to output the 'right' unit. ex, if the duration is more than one hour, it would make sense to convert that large second number to hour. etc. but this is for later PR.